staging/src/k8s.io/kubectl/pkg/cmd/config/config_test.go

```diff
@@ -364,24 +371,24 @@ func TestTokenAndBasicDisallowed(t *testing.T) {
 		expectedConfig: expectedConfig,
 	}
 
-	func() {
-		defer func() {
-			// Restore cmdutil behavior.
-			cmdutil.DefaultBehaviorOnFatal()
-		}()
-
-		// Check exit code.
-		cmdutil.BehaviorOnFatal(func(e string, code int) {
-			if code != 1 {
-				t.Errorf("The exit code is %d, expected 1", code)
-			}
-
-			expectedOutputs := []string{"--token", "--username"}
-			test.checkOutput(e, expectedOutputs, t)
-		})
-
+	// Allow for command errors in testing
+	if os.Getenv("CRASH") == "1" {
```

Comment:
I did not know that environment variable, is it something widely used in k8s and what does it for?

Reply:
This is a way to allow for commands to issue OS exit codes when they fail and keep testing them. Go doesn't really advise doing this, but it's possible, and I was struggling with the method used before for unit tests in these files, so I converted them to this method. What I intend to do is to just move these kinds of tests to the integration testing because that is generally what is advised these days for testing failure states like this. I'll need to do that in a separate PR as noted in my other comment.